pkg/util/k8scmcache.go

```diff
@@ -0,0 +1,208 @@
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"regexp"
+	"sync"
+
+        "github.com/golang/glog"
+        "github.com/pkg/errors"
+
+        k8s "k8s.io/client-go/kubernetes"
+        "k8s.io/client-go/rest"
+        "k8s.io/client-go/tools/clientcmd"
+        "k8s.io/api/core/v1"
+        metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type K8sCMCache struct {
+	Client *k8s.Clientset
+	Namespace string
+
+}
+
+const (
+        defaultNamespace = "default"
+
+        cmLabel = "csi-metadata"
+        cmDataKey = "content"
+
+        csiMetadataLabelAttr = "com.ceph.ceph-csi/persister"
+)
+
+var (
+	k8sCMCacheMtx sync.Mutex
+)
+
+func GetK8sNamespace() string {
+        namespace := os.Getenv("POD_NAMESPACE")
+        if namespace == "" {
+                return defaultNamespace
+        }
+        return namespace
+}
+
+func NewK8sClient() (*k8s.Clientset) {
+        var cfg *rest.Config
+        var err error
+        cPath := os.Getenv("KUBERNETES_CONFIG_PATH")
+        if cPath != "" {
+                cfg, err = clientcmd.BuildConfigFromFlags("", cPath)
+                if err != nil {
+                        glog.Errorf("Failed to get cluster config with error: %v\n", err)
+                        os.Exit(1)
+                }
+        } else {
+                cfg, err = rest.InClusterConfig()
+                if err != nil {
+                        glog.Errorf("Failed to get cluster config with error: %v\n", err)
+                        os.Exit(1)
+                }
+        }
+        client, err := k8s.NewForConfig(cfg)
+        if err != nil {
+                glog.Errorf("Failed to create client with error: %v\n", err)
+                os.Exit(1)
+        }
+        return client
+}
+
+func (k8scm *K8sCMCache) createMetadataCM(resourceID string) (*v1.ConfigMap, error) {
+        cm := &v1.ConfigMap{}
+        cm, err := k8scm.getMetadataCM(resourceID)
+        if err != nil {
+                glog.Infof("k8s-cm-cache: couldn't get configmap %s with error: %v", resourceID, err)
```

Comment:
This isn't _that_ important, let's use another verbosity level `glog.V(4).Infof(...)`. The same goes for other log messages in this method.